### PR TITLE
[13.0][NEW] Add new module crm_automated_activity

### PR DESCRIPTION
**Pull request by britoederr**
This new module allow create rules in CRM Stages in order to create automated activities when creating or moving opportunities in selected stages.

1. Go to stages form and mark 'Automated Activity for Stages'
2. Create the rules to be apply in this stage
3. Create or move an opportunity for this stage

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Available addons
 ----------------
 addon | version | summary
 --- | --- | ---
+[crm_automated_activity](crm_automated_activity/) | 13.0.1.0.1 | Automated Activities for CRM Stages
```
Comment (marcelsavegnago) [inline README.md]: You don't need to add it manually. This is updated by when the module is merged

File: crm_automated_activity/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "CRM Automated Activity",
+    "summary": """
+        Automated Activities for CRM Stages""",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "Eder Brito,Odoo Community Association (OCA)",
+    "website": "pingotecnologia.com.br",
```
Comment (marcelsavegnago) [inline crm_automated_activity/__manifest__.py]:
```suggestion
    "website": "https://github/oca/crm",
```

File: setup/crm_automated_activity/odoo/addons/crm_automated_activity [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../../crm_automated_activity
```
Comment (marcelsavegnago) [inline setup/crm_automated_activity/odoo/addons/crm_automated_activity:1]: As far as I know, you don't have to add this manually either.

File: setup/crm_automated_activity/setup.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import setuptools
```
Comment (marcelsavegnago) [inline setup/crm_automated_activity/setup.py:1]: As far as I know, you don't have to add this manually either.

**Review by marcelsavegnago — CHANGES_REQUESTED** [review]

File: crm_automated_activity/models/automated_activity.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 Eder Brito
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AutomatedActivity(models.Model):
+    _name = "automated.activity"
```
Comment (marcelsavegnago) [inline crm_automated_activity/models/automated_activity.py]: @rvalyi Is it better that the model name is crm.automated.activity?

File: README.md (hunk 1 above)
Comment (britoederr) [reply README.md]: Ok

File: setup/crm_automated_activity/odoo/addons/crm_automated_activity (hunk 3 above)
Comment (britoederr) [reply setup/crm_automated_activity/odoo/addons/crm_automated_activity:1]: Ok

File: crm_automated_activity/models/automated_activity.py (hunk 5 above)
Comment (britoederr) [reply crm_automated_activity/models/automated_activity.py]: @marcelsavegnago I changed this name to **crm.automated.activity**

**Comment by marcelsavegnago** [comment]
https://github.com/OCA/crm/pull/382/files#annotation_966701952

**Comment by britoederr** [comment]
@marcelsavegnago review please

File: crm_automated_activity/models/crm_lead.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    @api.model
+    def create(self, vals):
+
+        activity = self.env["mail.activity"]
+        crm_model = self.env["ir.model"].search([("model", "=", "crm.lead")])
+
+        res = super(CrmLead, self).create(vals)
+
+        if res.stage_id.create_automated_activity and any(
```
Comment (rousseldenis) [inline crm_automated_activity/models/crm_lead.py:21]: any is not necessary, just field is sufficent

File: crm_automated_activity/models/crm_lead.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            res.stage_id.automated_activity_ids
+        ):
+
+            for act in res.stage_id.automated_activity_ids:
+
+                deadline = date.today() + timedelta(days=act.days_deadline)
+
+                if act.apply_in in ["create", "create_write"]:
```
Comment (rousseldenis) [inline crm_automated_activity/models/crm_lead.py:29]: Could you put the activity creation in a function you can share between write and create ?

File: crm_automated_activity/models/crm_lead.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+                    activity.sudo().create(
+                        [
+                            {
+                                "res_id": res.id,
```
Comment (rousseldenis) [inline crm_automated_activity/models/crm_lead.py:34]: Could you put the values retrievement in a separate function in order to easy inheritance ?

File: crm_automated_activity/models/crm_lead.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+            res.stage_id.automated_activity_ids
+        ):
+
+            for act in res.stage_id.automated_activity_ids:
+
+                deadline = date.today() + timedelta(days=act.days_deadline)
+
+                if act.apply_in in ["create", "create_write"]:
+
+                    activity.sudo().create(
```
Comment (rousseldenis) [inline crm_automated_activity/models/crm_lead.py:31]: To improve performances, could you create a list of values and pass it to the create at the end (multi-create) ?

File: crm_automated_activity/readme/DESCRIPTION.rst [hunk 10]
```diff
@@ -0,0 +1,3 @@
+This application allows you automatically create activities for CRM Stages based in determined rules.
+
+A field is add to CRM Stage form in order to allow the activities creation
```
Comment (rousseldenis) [inline crm_automated_activity/readme/DESCRIPTION.rst:3]:
```suggestion
A field is added to CRM Stage form in order to allow the activities creation
```

File: crm_automated_activity/readme/USAGE.rst [hunk 11]
```diff
@@ -0,0 +1,3 @@
+Go to menu CRM > Configuration > Stages > Click in Create Automatically Activities.
+Create the pattern for the future activities in Automated Activities
```
Comment (rousseldenis) [inline crm_automated_activity/readme/USAGE.rst:2]: Could you give an example of pattern ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Thanks for this. Some remarks

**Comment by marcelsavegnago** [comment]
Hello @britoederr, how are you?

I think there is little left to complete this PR. Don't you want to continue?

**Pull request closed without merging**